src/test/java/uk/gov/hmcts/reform/civil/handler/callback/camunda/docmosis/GenerateResponseSealedSpecTest.java

```diff
@@ -0,0 +1,267 @@
+        CaseData updatedData = mapper.convertValue(response.getData(), CaseData.class);
+
+        // Then: updatedData should contain sealed form not stitched doc
+        assertThat(updatedData.getSystemGeneratedCaseDocuments().contains(SEALED_FORM));
+        verify(sealedClaimResponseFormGeneratorForSpec).generate(any(CaseData.class), eq(BEARER_TOKEN));
+    }
+
+    @Test
+    void shouldGenerateClaimForm_V1VersionAndIsPinInPostDisabledAndStitchDisabled() {
+        // Given: Case data with docs to stitch, stitching is disabled,isPinInPostEnabled is false and callback
+        // version V1
+        ReflectionTestUtils.setField(handler, "stitchEnabled", false);
+        CallbackParams params = callbackParamsOf(CallbackVersion.V_1, caseData, ABOUT_TO_SUBMIT);
+        when(toggleService.isPinInPostEnabled()).thenReturn(false);
+
+        // When: handler is called
+        var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+        CaseData updatedData = mapper.convertValue(response.getData(), CaseData.class);
+
+        // Then: updatedData should contain sealed form not stitched doc
+        assertThat(updatedData.getSystemGeneratedCaseDocuments().contains(SEALED_FORM));
+        verify(sealedClaimResponseFormGeneratorForSpec).generate(any(CaseData.class), eq(BEARER_TOKEN));
+    }
+
+    @Test
+    void shouldGenerateClaimForm_WhenSpecResponseTimelineDocAndRespondent1SpecDefenceResponseDocumentIsNull() {
+        // Given: Case data with docs to stitch and no existing systemGeneratedCaseDocuments,
+        // stitching is enabled,isPinInPostEnabled is false and callback version V1
+        CaseData caseData = CaseDataBuilder.builder()
+            .atStatePendingClaimIssued().build().toBuilder()
+            .specRespondent1Represented(YES)
+            .systemGeneratedCaseDocuments(new ArrayList<>()).build();
+        ReflectionTestUtils.setField(handler, "stitchEnabled", true);
+        CallbackParams params = callbackParamsOf(CallbackVersion.V_1, caseData, ABOUT_TO_SUBMIT);
+        when(toggleService.isPinInPostEnabled()).thenReturn(false);
+
+        // When: handler is called
+        var response = (AboutToStartOrSubmitCallbackResponse) handler.handle(params);
+        CaseData updatedData = mapper.convertValue(response.getData(), CaseData.class);
+
+        // Then: updatedData should contain stitched doc
+        assertThat(updatedData.getSystemGeneratedCaseDocuments().contains(STITCHED_DOC));
+        verify(sealedClaimResponseFormGeneratorForSpec).generate(any(CaseData.class), eq(BEARER_TOKEN));
+    }
+
+    @Test
+    void shouldReturnCorrectActivityId_whenRequestedSpecClaim() {
+        CaseData caseData = CaseDataBuilder.builder().atStatePendingClaimIssued().build();
+
+        CallbackParams params = callbackParamsOf(caseData, ABOUT_TO_SUBMIT);
+
+        assertThat(handler.handledEvents()).contains(CaseEvent.GENERATE_RESPONSE_SEALED);
+    }
+}
```

Comment:
Dead code in here. Lines 261 and 263 are not necessary.

```suggestion
    @Test
    void shouldReturnCorrectActivityId_whenRequestedSpecClaim() {
        assertThat(handler.handledEvents()).contains(CaseEvent.GENERATE_RESPONSE_SEALED);
    }
}
```